### PR TITLE
React

### PR DESCRIPTION
**Pull request by juliangruber**
Project: https://github.com/datproject/dat-desktop/projects/1

This is the initial code dump for the React port, from my private branch. See https://github.com/datproject/dat-desktop/issues/457 for motivation.

Next I'll organize and lay out in a new milestone and this issue everything left to be done, so we can properly collaborate on this and get it working asap.

Before any doubt arises, I want to thank highly @yoshuawuyts and the choo framework for serving us so well from inception of Dat-Desktop to now, growing and adapting as necessary :)

Currently the app looks like this:

screen shot 2018-01-27 at 13 02 47

screen shot 2018-01-27 at 13 00 38

screen shot 2018-01-27 at 13 00 43

### CODE REVIEW

**Comment by juliangruber** [comment]
I've created a GitHub project to track progress: https://github.com/datproject/dat-desktop/projects/1

File: app/actions/index.js [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+
+export const updateTemporaryTitleValue = title => ({
+  type: 'UPDATE_TEMPORARY_TITLE_VALUE',
+  title
+})
+
+export const updateTitle = (key, path, editValue) => async dispatch => {
+  const filePath = `${path}/dat.json`
+  const blob = await readFile(filePath)
+  const metadata = { ...JSON.parse(blob), title: editValue }
```
[the remark this answers is not in the document]
Comment (martinheidegger) [reply app/actions/index.js]: https://github.com/dat-land/dat-desktop/blob/5af70eb010b202d9e5af9d09a8909b92db3330db/lib/dat-json.js#L5

**Comment by soyuka** [comment]
I'd be to merge this to master and do a beta release, WDYT @martinheidegger ?

**Comment by karissa** [comment]
wow how exciting!

**Pull request merged**